apps/trackers/product_definition_handlers/fedramp_handler.py

```diff
@@ -0,0 +1,73 @@
+        True if FedrampHandler will modify the offers passed to get_offer. Can be used
+        by other handlers to avoid doing offers edits that would have to be reverted
+        by FedrampHandler.
+        """
+        return FedrampHandler.compliance_priority_fedramp_streams(affect, ps_module) and impact in FedrampHandler.FEDRAMP_OVERRIDES
+
+    @staticmethod
+    def compliance_priority_fedramp_streams(affect: Affect, ps_module: PsModule) -> list[PsUpdateStream]:
+        if affect.is_compliance_priority:
+            stream_names = CompliancePriority.objects.get(ps_module=ps_module.name).streams
+            streams = [PsUpdateStream.objects.get(name=name) for name in stream_names]
+            return streams
+
+    def get_offer(self, affect: Affect, impact: Impact, ps_module: PsModule, offers):
+        # compliance priority / fedramp special handling
+        #
+        # Only apply this special handling when fedramp stream is
+        # configured in the compliance_priority list.
+        if FedrampHandler.will_modify_offers(affect, impact, ps_module):
+
+            # If Fedramp streams are selected, the unacked stream must not be selected.
+            # Undo work done by unacked_handler.
+            # Ref. OSIDB-1876 comment #10 written by P.
+            unacked_stream = ps_module.unacked_ps_update_stream.first()
+            if unacked_stream:
+                offers[unacked_stream.name] = {
+                    "ps_update_stream": unacked_stream.name,
+                    "selected": False,
+                    "aus": False,
+                    "eus": False,
+                    "acked": False,
+                }
+
+            # If multiple fedramp streams are configured, only seed the
+            # streams_to_preselect with the first oldest one.
+            # TODO: If no good reason given, preselect all streams. But that might have high runtime complexity.
+            # TODO: Best solution probably: Update compliance_priority.yml to explicity state the first stream must be the oldest one.
+            stream_to_preselect = FedrampHandler.compliance_priority_fedramp_streams(affect, ps_module)[0]
+
+            # Moderate issues in fedramp components need to be fixed in
+            # fedramp streams and all following streams.
+            for stream in ps_module.default_ps_update_streams.all():
+                if ps_update_stream_natural_keys(stream) >= ps_update_stream_natural_keys(stream_to_preselect):
+                    offers[stream.name] = {
+                        "ps_update_stream": stream.name,
+                        "selected": True,
+                        "eus": bool(
+                            ps_module.eus_ps_update_streams.filter(name=stream.name)
+                        ),
+                        "aus": bool(
+                            ps_module.aus_ps_update_streams.filter(name=stream.name)
+                        ),
+                        "acked": not ps_module.unacked_ps_update_stream,
```

Comment:
I am not sure about this line or I do not understand it. It seems that it sets this stream as unacked whenever there exists an unacked stream for this module - which does not have to be the same stream.

Reply:
If I understand SFM2 right, `acked` is `True` for everything but the unacked stream. The unacted stream has it always `False` no matter whether it's selected or not. So perhaps a bug in OSIDB?

Reply:
Exactly as you describe. And it seems that it is probably bug in OSIDB.

Reply:
Fixed.